lib/TuyaOAuth2DriverLight.js

```diff
@@ -202,7 +202,7 @@ class TuyaOAuth2DriverLight extends TuyaOAuth2Driver {
     // Category Specifications
     // The main light category has both (0,255) and (0,1000) for backwards compatibility
     // Other categories use only (0,1000)
-    if (specifications.category === "dj") {
+    if (specifications && specifications.category === "dj") {
```

Comment:
```suggestion
    if (device.category === "dj") {
```
The device category in the specifications was used assuming they would always be present. This should also work if they are not and still give the right defaults.